style-guide: clarify placeholders (#15807)
tldr-pages/tldr: contributing-guides/style-guide.md

- If a command can optionally take 1 or more arguments of the same kind, use an ellipsis: `{{placeholder1 placeholder2 ...}}`.
  For instance, if multiple paths are expected, use `{{path/to/directory1 path/to/directory2 ...}}`.
- If only one of the multiple options is possible, write it as: `{{placeholder1|placeholder2|placeholder3}}`. If there are more than 4 possible values, you can use `|...` after the last item.

#### Optional placeholders

When documenting optional placeholders like paths or file extensions, it is suggested to specify them in the page or example descriptions instead of the placeholder itself. For example:

- Use `{{path/to/source.ext}}` instead of `{{path/to/source.tar[.gz|.bz2|.xz]}}`.

> > [!NOTE]\

Author: This is a QoL feature. See https://github.com/tldr-pages/tldr/pull/15802#discussion_r1971380265

> > Angle brackets `[]` must always be a substring of the placeholder to not get mixed up with an option placeholder.

Reviewer: ```suggestion
> Angle brackets `[]` must always be within the placeholder in order not to get mixed up with an option placeholder.
```

Author: This is not good enough. {{[string]}} is still within the placeholder

Reviewer: What exactly would you like to address with this PR? Could you give an example? I am not sure I understand what the problem is

Author: Since option placeholders will soon be a thing, {{[]}} will be reserved for those. To make sure pages don't use this pattern when long and short options are not involved.

Reviewer: I think we need to find a different wording then. In my opinion, "substring" implies that the literal sequence`[]` should appear (and not, e.g. `[something]`).

Author: I get what you mean. Please suggest something. I'm in bed and only responding to emails.

Author: I wasn't able to come up with a good rewording so I removed the note. Lets reconsider this if this becomes a problem.

Reviewer: Me too unfortunately
